tasks/bootstrap.yml

```diff
@@ -29,35 +29,63 @@
     - inventory_hostname == groups['openio_zk_cluster'][0]
       and openio_zk_status.rc != 0
 
+- name: "Gather PIDs of meta0/meta1 services"
+  shell: "{{ openio_gridinit_cmd }} status2 @meta0 @meta1 | grep {{openio_namespace}} | awk '{print $3}'"
+  register: m0m1_pid
+
+- name: "Restart meta0/meta1 services"
+  shell: "{{ openio_gridinit_cmd }} restart $({{ openio_gridinit_cmd }} status @meta0 @meta1 |
+          grep ' {{ openio_namespace }},' | awk '{print $1}')"
+
+- name: "Check that meta0/meta1 have restarted"
+  wait_for:
+    path: "/proc/{{ item }}/status"
+    state: absent
+  with_items: "{{ m0m1_pid.stdout_lines }}"
+  ignore_errors: yes
+  register: m0m1_pid_killed
+  timeout: 30
+
+- name: "Kill remaining meta0/meta1"
+  shell: "kill -9 {{ item }}"
+  with_items: "{{ m0m1_pid_killed.results | select('failed') | map(attribute='item') | list }}"
+
 - name: "Unlocking namespace {{ openio_namespace }}"
   command: "{{ openio_cli_path }} --oio-ns={{ openio_namespace }} cluster unlockall"
   when:
     - inventory_hostname == groups['openio_conscience'][0]
 
-- name: Waiting for scores to increase
-  pause:
-    seconds: 10
+- name: "Waiting for {{  groups['openio_directory_m0'] | length }} meta0 and {{ groups['openio_directory_m1'] | length }}
+         meta1 scores to increase"
+  shell: "{{ openio_cli_path }} cluster list meta0 meta1 --oio-ns={{ openio_namespace }}
+          -f value -c Score | grep -c '^0$' || true"
+  register: metas_count
+  until: "(metas_count.stdout | int) == 0"
+  retries: 10
+  delay: 1
+  when:
+    - inventory_hostname == groups['openio_conscience'][0]
 
 - name: "Bootstrapping OpenIO namespace {{ openio_namespace }} with {{ openio_replicas }} replicas"
   command: "{{ openio_cli_path }} --oio-ns={{ openio_namespace }} directory bootstrap --replicas {{ openio_replicas }} --no-rdir"
   when:
     - inventory_hostname == groups['openio_conscience'][0]
 
-- name: Update meta0 services
-  command: "{{ openio_gridinit_cmd }} restart @meta0"
-  when:
-    - inventory_hostname in groups['openio_directory_m0']
-
-- name: Update meta1 services
-  command: "{{ openio_gridinit_cmd }} restart @meta1"
+- name: "Update meta0/meta1 services"
+  command: "{{ openio_gridinit_cmd }} restart @meta0 @meta1"
```

Comment:
Are you sure this will work if m0 & m1 server groups are not the same ?

What's the rationale to restart them at the same time ?